### PR TITLE
Add LogStream

### PR DESCRIPTION
**Pull request by edbaunton**
Add the logstream API, originally pushed upstream by Eric Burnett to
gist.github.com/EricBurnett/a094e0b8474b834d098e86bf20fa202b via
groups.google.com/g/remote-execution-apis/c/LCLsBSgSnU0/m/yTTtqr-cAwAJ

Update to include a copyright notice; set the version number to v1 and update
the package to reflect the location within the remote-apis repo.

I made no material changes to the logstream api other than:
* to prefix the proto and namespace to be consistent with the others found in this repo, so that it is `remote_logstream.proto`.
* Add a copyright notice
* Setup the namespaces in the proto mirroring what was done for the asset api

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
-cp "$(find $(bazel info bazel-bin) -path "*/$GEN_REPB_PATH")" "$GEN_REPB_PATH"
-GEN_SEMVERPB_PATH="build/bazel/semver/semver.pb.go"
-cp "$(find $(bazel info bazel-bin) -path "*/$GEN_SEMVERPB_PATH")" "$GEN_SEMVERPB_PATH"
-GEN_RFPB_PATH="build/bazel/remote/asset/v1/remote_asset.pb.go"
-cp "$(find $(bazel info bazel-bin) -path "*/$GEN_RFPB_PATH")" "$GEN_RFPB_PATH"
-git add "$GEN_REPB_PATH" "$GEN_SEMVERPB_PATH" "$GEN_RFPB_PATH"
+      build/bazel/remote/asset/v1:remote_asset_go_proto \
+      build/bazel/remote/logstream/v1:remote_logstream.proto
+
+for i in "build/bazel/remote/execution/v2/remote_execution.pb.go" \
```
Comment (sstriker) [inline hooks/pre-commit:9]: I like this generalization / simplification.

File: build/bazel/remote/logstream/v1/remote_logstream.proto [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Log Stream API
+
+syntax = "proto3";
+
+package build.bazel.remote.logstream.v1;
+
+option csharp_namespace = "Build.Bazel.Remote.Logstream.v1";
```
Comment (EricBurnett) [inline build/bazel/remote/logstream/v1/remote_logstream.proto:21]: Should it be 'LogStream' in the csharp/java namespaces?

File: build/bazel/remote/logstream/v1/remote_logstream.proto [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  // The list of valid types of parent resources of LogStreams is up to the
+  // implementing server.
+  // Example: projects/123
+  string parent = 1;
+}
+
+// A handle to a log (an ordered sequence of bytes).
+message LogStream {
+  // Structured name of the resource in the format:
+  //   {parent=**}/logStreams/{logstream_id}
```
Comment (EricBurnett) [inline build/bazel/remote/logstream/v1/remote_logstream.proto:113]: Let's correct this while we're here: logStreams -> logstreams. (Resources are usually lowercase, and this is already wrong with having a lowercase l and uppercase S)

Comment (EricBurnett) [reply build/bazel/remote/logstream/v1/remote_logstream.proto]: Couple more: lines 114, 120, and 121 also thanks!

Comment (edbaunton) [reply build/bazel/remote/logstream/v1/remote_logstream.proto]: Ah good catch! Should be fixed now.

**Pull request merged**